lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb

```diff
@@ -111,16 +134,28 @@ def validate_presence_of(attr)
 
       # @private
       class ValidatePresenceOfMatcher < ValidationMatcher
+        def initialize(attribute)
+          super(attribute)
+          @options = {}
+        end
+
         def with_message(message)
           @expected_message = message if message
           self
         end
 
+        def allow_nil(allow_nil = true)
+          @options[:allow_nil] = allow_nil
+          self
+        end
+
         def matches?(subject)
           super(subject)
           @expected_message ||= :blank
 
-          if secure_password_being_validated?
+          if @options[:allow_nil]
```

Comment:
Using an `elsif` here will make it so that we no longer check that the attribute is invalid with a blank value, and I think we still need that.

To make it so that we don't contradict ourselves when `allow_nil` is specified, instead of the `blank_value` method below returning `nil`, what if it returns an empty string?